regolith/helpers/u_todohelper.py

```diff
@@ -96,51 +103,81 @@ def db_updater(self):
+                if len(todolist_idx) == 0:
+                    continue
+                else:
+                    for todo in todolist_idx:
+                        if todo.get('status') == "started":
+                            todo["running_index"] = index
+                            index += 1
+                        if todo.get('status') in ["finished", "cancelled"]:
+                            todo["running_index"] = index_finished
+                            index_finished += -1
+        person = document_by_value(all_docs_from_collection(rc.client, "people"), "_id", rc.assigned_to)
         if not person:
             raise TypeError(f"Id {rc.assigned_to} can't be found in people collection")
         todolist = person.get("todos", [])
         if len(todolist) == 0:
             print(f"{rc.assigned_to} doesn't have todos in people collection.")
             return
-        index = 1
-        for todo in todolist:
-            if todo.get('status') == "started":
-                todo["index"] = index
-                index += 1
-        for todo in todolist:
-            if todo.get('status') in ["finished", "cancelled"]:
-                todo["index"] = index
-                index += 1
         if not rc.index:
-            print("-" * 50)
+            if not rc.certain_date:
+                today = dt.date.today()
+            else:
+                today = date_parser.parse(rc.certain_date).date()
+            for todo in todolist:
+                if not todo.get('importance'):
+                    todo['importance'] = 1
+                if type(todo["due_date"]) == str:
+                    todo["due_date"] = date_parser.parse(todo["due_date"]).date()
+                todo["days_to_due"] = (todo.get('due_date') - today).days
+                todo["order"] = todo['importance'] + 1 / (1 + math.exp(abs(todo["days_to_due"])))
+            todolist = sorted(todolist, key=lambda k: (-k['order'], k.get('duration', 10000), k['status']))
+            index_match={}
+            if rc.running_index:
+                new_index = 1
+                for todo in todolist:
+                    if todo["status"] == 'started':
+                        index_match[todo["running_index"]] = new_index
+                        new_index += 1
+                for i in range(0, len(rc.databases)):
+                    db_name = rc.databases[i]["name"]
+                    person_idx = rc.client.find_one(db_name, rc.coll, filterid)
+                    todolist_idx = person_idx.get("todos", [])
+                    if len(todolist_idx) != 0:
+                        for todo in todolist_idx:
+                            if todo.get('status') == "started":
+                                index = index_match[todo["running_index"]]
+                                todo["running_index"] = index
+                        rc.client.update_one(db_name, rc.coll, {'_id': rc.assigned_to}, {"todos": todolist_idx},
```

Comment:
Then, pair the temporary index with the updated order, and saved them in index_match. 
Finally, update tasks again..

The logic looks complex, while I couldn't think up an easier one... At least these codes work fine currently!